glue/plugins/tools/spectrum_tool/qt/spectrum_tool.py

```diff
@@ -68,18 +70,6 @@ def spectrum(data, attribute, roi, slc, zaxis):
             data = data[attribute, tuple(slc)]   # attribute is Primary,
             finite = np.isfinite(data)
 
-            assert data.ndim == ndim
-
-            for i in reversed(list(range(ndim))):
-                if i != zaxis:  # index of the zaxis in dataset
-                    data = np.nansum(data, axis=i)
-                    finite = finite.sum(axis=i)
-
-            assert data.ndim == 1
-            assert data.size == nz
-
-            data = (1. * data / finite).ravel()
-
```

Comment:
Is it normal that you removed all this? Does it not break rectangular spectrum extraction?